config/pg_types.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Overwrite node-pg-types date parser. See https://github.com/tgriesser/knex/issues/1750
+const pgTypes = require('pg').types;
+
+const DATE_OID = 1082;
+const DATE_ARRAY_OID = 1182;
+
+function parseDate(value) {
+  if (!value) {
+    return null;
+  }
+
+  return new Date(Date.parse(value));
+}
+
+function parseDateArray(value) {
+  if (!value) {
+    return null;
+  }
+
+  const p = pgTypes.arrayParser.create(value, (entry) => {
+    let parsedEntry = entry;
+    if (String(entry).toLowerCase() !== String(null)) {
+      parsedEntry = parseDate(entry);
+    } else {
+      parsedEntry = null;
+    }
+    return parsedEntry;
+  });
+
+  return p.parse();
```

Comment:
A bit unclear what `p` means, unless you go back up to see it's a parser. Might be worth renaming it to `parser`.